include/aws/checksums/exports.h

```diff
@@ -14,13 +14,12 @@
 #    else
 #        define AWS_CHECKSUMS_API
 #    endif /* AWS_CHECKSUMS_USE_IMPORT_EXPORT */
-#else      /* defined (AWS_C_RT_USE_WINDOWS_DLL_SEMANTICS) || defined (_WIN32) */
-#    if ((__GNUC__ >= 4) || defined(__clang__)) && defined(AWS_CHECKSUMS_USE_IMPORT_EXPORT) &&                         \
-        defined(AWS_CHECKSUMS_EXPORTS)
+#else      /* defined (AWS_CRT_USE_WINDOWS_DLL_SEMANTICS) || defined (_WIN32) */
+#    if defined(AWS_CHECKSUMS_USE_IMPORT_EXPORT) && defined(AWS_CHECKSUMS_EXPORTS)
```

Comment:
I don't 100% get why we made this change. Was this a bug where we end up not defining anything in the edge case? Maybe update the PR description so that we know why we made this change in the future. As a side note, we were recently discussing windows with clang, does `defined(_WIN32)` check work correctly in that case?

Reply:
its not a bug, just very inconsistent across the codebase, so making it a bit more consistent and removing old cruft. will add comment to description.